htlcswitch/hop/iterator.go

```diff
@@ -198,34 +199,34 @@ func (b *BlindingKit) validateBlindingPoint(payloadBlinding *btcec.PublicKey,
 
 	switch {
 	case !(payloadBlindingSet || updateBlindingSet):
-		return nil, ErrInvalidPayload{
+		return nil, 0, ErrInvalidPayload{
 			Type:      record.BlindingPointOnionType,
 			Violation: OmittedViolation,
 			FinalHop:  isFinalHop,
 		}
 
 	case payloadBlindingSet && updateBlindingSet:
-		return nil, ErrInvalidPayload{
+		return nil, 0, ErrInvalidPayload{
 			Type:      record.BlindingPointOnionType,
 			Violation: IncludedViolation,
 			FinalHop:  isFinalHop,
```

Comment:
ah yes, I think I was referring to the return value as you described, but potentially one could set the role in `ErrInvalidPayload`, what if the sender constructs an onion with `current_blinding` inside a blinded path?

Reply:
> what if the sender constructs an onion with current_blinding inside a blinded path

Good point, if payload and `update_add` blinding are both set how do we handle that error? Because we don't know which role we're expected to play (since we've either got a buggy blinded route that incorrectly added to payload or a buggy peer that incorrectly sent in `update_add`).

Seems like the safest behavior to act as if we're the introduction node so that we correctly convert the error as expected by mission control? Will ask what others do, but that's also what you'd do if you just read the spec [in the order it's written...](https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#requirements-25)